Mathlib/Data/List/Infix.lean

```diff
@@ -70,6 +70,18 @@ theorem infix_rfl : l <:+: l :=
 theorem prefix_concat (a : α) (l) : l <+: concat l a := by simp
 #align list.prefix_concat List.prefix_concat
 
+theorem prefix_concat_iff {x y : List α} {a : α} : x <+: y ++ [a] ↔ x <+: y ∨ x = y ++ [a] := by
```

Comment:
Curious question: Can you easily deduce this by putting some `reverse` everywhere and applying `List.suffix_cons_iff`?

Reply:
Indeed